### PR TITLE
fix(Makefile): adopt makefile for docker-compose.yml

### PR DESCRIPTION
**Pull request by noasck**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,12 @@
 FROM golang:buster
-ARG GOPROXY=https://goproxy.cn,direct
 
-COPY ./src /app
+COPY ./src/go.mod /app/
+COPY ./src/go.sum /app/
 WORKDIR /app
+
 RUN go mod download
```
Comment (noasck) [inline Dockerfile:7]: With these changes, Docker won't download requirements every time we change the code

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,4 @@
-.PHONY: build
-build:
-	go build -v ./src
-
-.DEFAULT_GOAL := build
\ No newline at end of file
+dev:
+	docker-compose -f docker-compose.yml up --build
+down:
+	docker-compose -f docker-compose.yml down --volumes --remove-orphans
```
Comment (noasck) [inline Makefile:4]: Should work on windows

File: src/main.go [hunk 3]
```diff
@@ -14,14 +14,15 @@ import (
...
+	app := gin.New()
+	app.Use(gin.Recovery())
 
 	// Config Init
 	conf := config.Config(".")
 	log.Info("Config set up successfully!")
 
 	// Logger init
-	app.Use(utils.Logger_JSON(conf.Log.Filename, true))
+	app.Use(utils.Logger_JSON(conf.Log.Filename, conf.Gin.Mode))
```
Comment (noasck) [inline src/main.go:25]: Handling logs depending on Gin Mode

File: src/utils/access_logger.go [hunk 4]
```diff
@@ -8,25 +8,31 @@ import (
...
-		DisableColors: w_stdout,
+		DisableColors: w_file,
 		FullTimestamp: true,
 	})
 
-	if w_stdout {
+	if w_file {
 		log.SetLevel(log.InfoLevel)
-
+		log.SetReportCaller(true)
```
Comment (noasck) [inline src/utils/access_logger.go:22]: To attach call log file to log message

**Review by 2upo — APPROVED** [review]

**Pull request merged**